### PR TITLE
[Enhancement] refactor allin1 image

### LINKED ISSUE #13319: Stream load Parquet file
I imagine it's a lot more performant, given the bandwidth would be lower, the types are pre-defined, and the columnar storage is similar.

### PR DESCRIPTION
**Pull request by kevincai**
* add supervisor and nginx into allin1
* add broker into allin1
* robust register BE BROKER into FE
* be able to use stream load directly talking to nginx http port

Fixes #13319

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.1
  - [X] 3.0
  - [X] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docker/dockerfiles/allin1/services/director/run.sh [hunk 1]
```diff
@@ -0,0 +1,236 @@
+#!/bin/bash
```
Comment (imay) [inline docker/dockerfiles/allin1/services/director/run.sh:1]: Better to add license header for these new added files.

**Review by imay — COMMENTED** [review]
Wonderful job, how much additional volume are added to the image size after this change?

**Comment by kevincai** [comment]
> 

about 450MB fs_broker size, I guess around 100-200MB additional storage after compress.

File: docker/dockerfiles/allin1/services/director/run.sh (hunk 1 above)
Comment (kevincai) [reply docker/dockerfiles/allin1/services/director/run.sh:1]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docker/dockerfiles/allin1/services/director/run.sh [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        sleep $SQL_RETRY_INTERNAL
+    done
+}
+
+check_and_add_be()
+{
+    loginfo "check if need to add BE into FE service ..."
+    while true
+    do
+        result=`exec_sql_with_retry "SHOW BACKENDS;"`
```
Comment (kevincai) [inline docker/dockerfiles/allin1/services/director/run.sh:175]: `result` can be possibly mixtured with error messages, need fix it here.

**Review by imay — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docker/dockerfiles/allin1/allin1-ubi.Dockerfile [hunk 3]
```diff
@@ -29,25 +30,25 @@ FROM registry.access.redhat.com/ubi8/ubi:8.7
 ARG DEPLOYDIR=/data/deploy
 ENV SR_HOME=${DEPLOYDIR}/starrocks
 
-RUN yum install -y java-1.8.0-openjdk-devel tzdata openssl curl vim ca-certificates fontconfig gzip tar less hostname procps-ng lsof && \
-    rpm -ivh https://repo.mysql.com/mysql57-community-release-el7.rpm && \
+RUN yum install -y java-1.8.0-openjdk-devel tzdata openssl curl vim ca-certificates fontconfig gzip tar less hostname procps-ng lsof python3-pip nginx nc && \
+    rpm -ivh https://repo.mysql.com//mysql80-community-release-el8-7.noarch.rpm && \
```
Comment (yandongxiao) [inline docker/dockerfiles/allin1/allin1-ubi.Dockerfile:34]: two slash

File: docker/dockerfiles/be/be-ubi.Dockerfile [hunk 4]
```diff
@@ -28,9 +28,9 @@ FROM registry.access.redhat.com/ubi8/ubi:8.7
 ARG STARROCKS_ROOT=/opt/starrocks
 
 RUN yum install -y java-1.8.0-openjdk-devel tzdata openssl curl vim ca-certificates fontconfig gzip tar less hostname procps-ng lsof && \
-    rpm -ivh https://repo.mysql.com/mysql57-community-release-el7.rpm && \
+    rpm -ivh https://repo.mysql.com//mysql80-community-release-el8-7.noarch.rpm &&  \
```
Comment (yandongxiao) [inline docker/dockerfiles/be/be-ubi.Dockerfile:31]: double slash

File: docker/dockerfiles/fe/fe-ubi.Dockerfile [hunk 5]
```diff
@@ -28,9 +28,9 @@ FROM registry.access.redhat.com/ubi8/ubi:8.7
 ARG STARROCKS_ROOT=/opt/starrocks
 
 RUN yum install -y java-1.8.0-openjdk-devel tzdata openssl curl vim ca-certificates fontconfig gzip tar less hostname procps-ng lsof && \
-    rpm -ivh https://repo.mysql.com/mysql57-community-release-el7.rpm && \
+    rpm -ivh https://repo.mysql.com//mysql80-community-release-el8-7.noarch.rpm && \
```
Comment (yandongxiao) [inline docker/dockerfiles/fe/fe-ubi.Dockerfile:31]: double slash

File: docker/dockerfiles/allin1/services/director/run.sh [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+loginfo
+loginfo
+loginfo "$PERSISTENT_MSG"
+loginfo
+loginfo "FE mysql query port: `fe_mysql_port`"
+loginfo "FE http service port: 8080"
+loginfo
+loginfo "Enjoy the journal to StarRocks blazing-fast lake-house engine!"
+
+while true
```
Comment (yandongxiao) [inline docker/dockerfiles/allin1/services/director/run.sh:238]: this `while loop` is used to make run.sh run as a service, to avoid to exit?

Comment (kevincai) [reply docker/dockerfiles/allin1/services/director/run.sh:238]: yes

File: docker/dockerfiles/fe/fe-ubi.Dockerfile (hunk 5 above)
Comment (kevincai) [reply docker/dockerfiles/fe/fe-ubi.Dockerfile]: done

File: docker/dockerfiles/be/be-ubi.Dockerfile (hunk 4 above)
Comment (kevincai) [reply docker/dockerfiles/be/be-ubi.Dockerfile]: done

File: docker/dockerfiles/allin1/allin1-ubi.Dockerfile (hunk 3 above)
Comment (kevincai) [reply docker/dockerfiles/allin1/allin1-ubi.Dockerfile]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by yandongxiao — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**